nucliadb_vectors/src/data_point_provider/mod.rs

```diff
@@ -254,6 +262,62 @@ impl Index {
             }
         }
     }
+
+    fn take_available_merge_or_wait(&mut self) -> Option<Journal> {
+        let MergerStatus::WorkScheduled(rcv) = std::mem::take(&mut self.merger_status) else {
+            return None;
+        };
+        rcv.recv().ok()
+    }
+
+    /// Returns the number of segments that have been merged.
+    pub fn force_merge(&mut self, _lock: &Lock) -> VectorR<ForceMergeMetrics> {
+        let possible_merge = self.take_available_merge_or_wait();
+        let mut state = self.write_state();
+        let mut date = self.write_date();
+
+        if let Some(journal) = possible_merge {
+            state.replace_work_unit(journal);
+            fs_state::persist_state::<State>(&self.location, &state)?;
+            *date = fs_state::crnt_version(&self.location)?;
+        }
+
+        if state.work_stack_len() <= 1 {
+            return Ok(ForceMergeMetrics {
+                merged: 0,
+                segments_left: state.work_stack_len(),
+            });
+        }
+
+        let channel = self.metadata.channel;
+        let location = self.location.clone();
+        let similarity = self.metadata.similarity;
+        let force_merge_capacity = 100;
+        let mut live_segments: Vec<_> = state.dpid_iter().collect();
+        let mut buffer = Vec::with_capacity(force_merge_capacity);
+
+        while buffer.len() < force_merge_capacity {
+            let Some(journal) = live_segments.pop() else {
+                break;
+            };
+            buffer.push((state.delete_log(journal), journal.id()));
+        }
```

Comment:
Maybe we should limit by number of nodes per segment?

```suggestion
        let force_merge_capacity = 50_000;
        let mut live_segments: Vec<_> = state.dpid_iter().collect();
        let mut buffer = Vec::with_capacity(force_merge_capacity);
        let mut nodes_to_merge = 0;

        while nodes_to_merge < force_merge_capacity {
            let Some(journal) = live_segments.pop() else {
                break;
            };
            if journal.no_nodes() > force_merge_capacity {
              live_segments.push(journal);
            } else {
              buffer.push((state.delete_log(journal), journal.id()));
              nodes_to_merge += journal.no_nodes();
            }
        }
```

Reply:
Seems like another good metric, definitely useful, however to make the right decision we should get more metrics right? I went with this because testing is easier tbh...

Reply:
If a segment is big enough this is an infinite loop though:
```rust
while nodes_to_merge < force_merge_capacity {
            let Some(journal) = live_segments.pop() else {
                break;
            };
            if journal.no_nodes() > force_merge_capacity {
              live_segments.push(journal);
            } else {
              buffer.push((state.delete_log(journal), journal.id()));
              nodes_to_merge += journal.no_nodes();
            }
}
```

Reply:
yeah, I somehow thought `live_segments` was two separate vecs! :sweat:

Reply:
Threshold added!